Sources/ProjectDescription/SettingsTransformers.swift

```diff
@@ -0,0 +1,137 @@
+import Foundation
+
+public extension SettingsDictionary {
+    mutating func merge(_ other: SettingsDictionary) {
+        merge(other) { $1 }
+    }
+
+    func merging(_ other: SettingsDictionary) -> SettingsDictionary {
+        merging(other) { $1 }
+    }
+}
+
+private extension SettingValue {
+    init(_ string: String) {
+        self = .init(stringLiteral: string)
+    }
+
+    init(_ bool: Bool) {
+        self = .init(booleanLiteral: bool)
+    }
+}
+
+public enum SwiftCompilationMode: String {
+    case singlefile
+    case wholemodule
+}
+
+public enum SwiftOptimizationLevel: String {
+    case o = "-O"
+    case oNone = "-Onone"
+    case oSize = "-Osize"
+}
+
+public extension SettingsDictionary {
+    // MARK: - Deployment target and SDKROOT
+
+    /// Sets "SDKROOT" to `value`
+    func sdkRoot(_ value: String) -> SettingsDictionary {
+        merging(["SDKROOT": SettingValue(value)])
+    }
+
+    // MARK: - Code signing
+
+    /// Sets "CODE_SIGN_STYLE" to "Manual"
+    func manualCodeSigning(identity: String? = nil, provisioningProfileSpecifier: String? = nil) -> SettingsDictionary {
+        var manualCodeSigning: SettingsDictionary = ["CODE_SIGN_STYLE": "Manual"]
+        manualCodeSigning["PROVISIONING_PROFILE_SPECIFIER"] = provisioningProfileSpecifier.map { SettingValue($0) }
+
+        let merged = merging(manualCodeSigning)
+
+        guard let identity = identity else {
+            return merged
+        }
+
+        return merged.codeSignIdentity(identity)
+    }
+
+    /// Sets "CODE_SIGN_STYLE" to "Automatic" and "DEVELOPMENT_TEAM" to `devTeam`
+    func automaticCodeSigning(devTeam: String) -> SettingsDictionary {
+        merging([
+            "CODE_SIGN_STYLE": "Automatic",
+            "DEVELOPMENT_TEAM": SettingValue(devTeam),
+        ])
+    }
+
+    /// Sets "CODE_SIGN_IDENTITY" to "Apple Development"
+    func codeSignIdentityAppleDevelopment() -> SettingsDictionary {
+        codeSignIdentity("Apple Development")
+    }
+
+    /// Sets "CODE_SIGN_IDENTITY" to `identity`
+    func codeSignIdentity(_ identity: String) -> SettingsDictionary {
+        merging(["CODE_SIGN_IDENTITY": SettingValue(identity)])
+    }
+
+    // MARK: - Versioning and Product Name
+
+    /// Sets "PRODUCT_NAME" to `name`
+    func productName(_ name: String) -> SettingsDictionary {
+        merging(["PRODUCT_NAME": SettingValue(name)])
+    }
```

Comment:
I appreciate this is an optional helper, however it may be best to avoid exposing this as part of the Settings public API seeing Tuist already offers an API for it via `Target.productName`.

The reasoning is, setting this via build setting can yield some unexpected errors in the event that the product name mismatches what Tuist thinks the product name is. The `productName` is use extensively when generating dependencies between targets.

e.g.

![Screen Shot 2020-04-20 at 6 59 44 AM](https://user-images.githubusercontent.com/11914919/79718698-8be5eb00-82d4-11ea-9de2-f7968c246211.png)

The product names from `Framework1` and `Framework2` are used to derive those ^ 🙂

https://github.com/tuist/tuist/blob/master/Sources/TuistCore/Models/Target.swift#L93

Reply:
Good to know, thanks Kas!